lib/java/com/google/android/material/slider/BaseSlider.java

```diff
@@ -354,6 +354,8 @@ abstract class BaseSlider<
   private final ViewTreeObserver.OnGlobalLayoutListener onGlobalLayoutListener =
       this::updateLabels;
 
+  private boolean isVisible = isShown();
```

Comment:
actually I'm thinking we can do `isVisible = isShown()` in the constructor after the `super()` call (that way the super class has at least processed the initial setVisibility()), and then also `isVisible = isShown()` in `onAttachedToWindow()` to account for the parent hierarchy.

Reply:
I was able to run [Test8](https://github.com/material-components/material-components-android/issues/4319#issue-2554749506) without any problems in the following cases:
- BottomAppBar
  - android:visibility="visible"
  - android:visibility="invisible"
  - android:visibility="gone"

https://docs.oracle.com/javase/specs/jls/se23/html/jls-12.html#jls-12.5

Reply:
I modified the initialization of isVisible.

Reply:
BottomAppBar? The case I'm referring to is when the parent of the Slider is invisible/gone, isShown() won't work correctly in the constructor because the parent hierarchy is not assigned yet:

https://stackoverflow.com/questions/7099249/using-view-getparent-in-views-constructor

I'll add it to `onAttachedToWindow()` when importing the PR.

Reply:
Thank you very much. I see.

I was able to run [Test8](https://github.com/material-components/material-components-android/issues/4319#issue-2554749506) without any problems in the following cases:
```
  <com.google.android.material.bottomappbar.BottomAppBar
    android:id="@+id/appBar"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:layout_gravity="bottom"
    android:fitsSystemWindows="true"
    android:visibility="visible">				<=
    <com.google.android.material.slider.Slider
      android:id="@+id/slider"
      android:layout_width="match_parent"
      android:layout_height="wrap_content"
      app:labelBehavior="visible" />
  </com.google.android.material.bottomappbar.BottomAppBar>
```
```
  <com.google.android.material.bottomappbar.BottomAppBar
    android:id="@+id/appBar"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:layout_gravity="bottom"
    android:fitsSystemWindows="true"
    android:visibility="invisible">				<=
    <com.google.android.material.slider.Slider
      android:id="@+id/slider"
      android:layout_width="match_parent"
      android:layout_height="wrap_content"
      app:labelBehavior="visible" />
  </com.google.android.material.bottomappbar.BottomAppBar>
```
```
  <com.google.android.material.bottomappbar.BottomAppBar
    android:id="@+id/appBar"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:layout_gravity="bottom"
    android:fitsSystemWindows="true"
    android:visibility="gone">					<=
    <com.google.android.material.slider.Slider
      android:id="@+id/slider"
      android:layout_width="match_parent"
      android:layout_height="wrap_content"
      app:labelBehavior="visible" />
  </com.google.android.material.bottomappbar.BottomAppBar>
```